tpm/constants.go

```diff
@@ -110,23 +112,38 @@ const (
 	pidTransport
 )
 
+// Algorithm type for more convenient handling.
+// see Algorithm ID for possible values.
+type Algorithm uint32
+
 // Algorithm ID values.
 const (
-	_ uint32 = iota
-	algRSA
+	_ Algorithm = iota
+	AlgRSA
 	_ // was DES
 	_ // was 3DES in EDE mode
-	algSHA
-	algHMAC
-	algAES128
-	algMGF1
-	algAES192
-	algAES256
-	algXOR
+	AlgSHA
+	AlgHMAC
+	AlgAES128
+	AlgMGF1
+	AlgAES192
+	AlgAES256
+	AlgXOR
 )
 
+// AlgMap Map of Algorithms to Strings for nicer output and comparisons, etc.
+var AlgMap = map[Algorithm]string{
```

Comment:
This should still be hidden via a String method:
```
var algNames = map[Algorithm]string{
	AlgRSA:    "RSA",
	AlgSHA:    "SHA1",
	AlgHMAC:   "HMAC",
	AlgAES128: "AER128",
	AlgMGF1:   "MFG1",
	AlgAES192: "AES192",
	AlgAES256: "AES256",
}

func (a Algorithm) String() string {
  n, ok := algNames[a]
  if !ok {
    return "unknown_algorithm"
  }
  return n
}

Reply:
Hey, don't worry :) im really thankful for the reviewing and nit-picking is good, because I can improve my coding. :+1: